server/mdm/apple/apple_mdm.go

```diff
@@ -660,6 +660,18 @@ func (d *DEPService) processDeviceResponse(
 	for _, device := range addedDevicesSlice {
 		addedSerials = append(addedSerials, device.SerialNumber)
 	}
+
+	// Check if any of the "added" or "modified" hosts are hosts that we've recently removed from
+	// Fleet in ABM. A host in this state will have a row in `host_dep_assignments` where the
+	// `deleted_at ` col is NOT NULL. Down below we skip assigning the profile to devices that we
+	// think are still enrolled; doing this check here allows us to avoid skipping devices that
+	// _seem_ like they're still enrolled but were actually removed and should get the profile.
+	// See https://github.com/fleetdm/fleet/issues/23200 for more context.
+	existingDeletedSerials, err := d.ds.GetMatchingHostSerialsMarkedDeleted(ctx, addedSerials)
```

Comment:
What if the serial was added to a different team? Like the device changed ABM tokens?

Reply:
@getvictor could you expand more on that?

I think this method wouldn't be impacted by/impact a device changing to a team with a different ABM token, but I can test that out more thoroughly.

Reply:
Okidokie, these are the steps I followed:

1. Enroll in Fleet via ADE with the regular Fleet ABM token
2. Turn off MDM on the device
3. Unassign the device from Fleet ABM
4. Release the device from Fleet ABM
5. Reset the device
6. Use Apple Configurator to add the device to the Millrule Timbers ABM
7. Assign the device to my dev Fleet instance in the Millrule Timbers ABM
8. Transfer the device to another Fleet team that is set as the Millrule ABM token default macOS team
9. Finish setup and enroll the device

The ADE flow worked without a hitch! The device's `host_dep_assignment` was overwritten with the new one, which I expected.

@getvictor @gillespi314